### PR TITLE
added AttestationAuthority crd to kritis

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

**Comment by aaron-prindle** [comment]
for testing CRDs, is there a plan in place or are the integration tests sufficient here?

**Comment by dlorenc** [comment]
```shell
./hack/../pkg is out of date. Please run hack/update-codegen.sh
```

File: pkg/kritis/apis/kritis/v1beta1/register.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 	scheme.AddKnownTypes(SchemeGroupVersion,
 		&ImageSecurityPolicy{},
 		&ImageSecurityPolicyList{},
+		&AttestationAuthority{},
```
Comment (tejal29) [inline pkg/kritis/apis/kritis/v1beta1/register.go]: Shd we not add AttestationAuthorityList here as well?

File: artifacts/attestation-authority-example.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: kritis.grafeas.io/v1beta1
```
Comment (tejal29) [inline artifacts/attestation-authority-example.yaml]: Maybe move this to artifacts/examples/ ?

File: pkg/kritis/apis/kritis/v1beta1/register.go (hunk 1 above)
Comment (aaron-prindle) [reply pkg/kritis/apis/kritis/v1beta1/register.go]: done

File: artifacts/attestation-authority-example.yaml (hunk 2 above)
Comment (aaron-prindle) [reply artifacts/attestation-authority-example.yaml]: done

File: pkg/kritis/apis/kritis/v1beta1/types.go [hunk 3]
```diff
@@ -54,3 +54,26 @@ type ImageSecurityPolicyList struct {
...
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AttestationAuthority struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	NoteReference        []string `json:"noteReference"`
```
Comment (dlorenc) [inline pkg/kritis/apis/kritis/v1beta1/types.go]: why are these string slices? The example shows a single string.

Comment (aaron-prindle) [reply pkg/kritis/apis/kritis/v1beta1/types.go]: fixed

File: pkg/kritis/apis/kritis/v1beta1/types.go [hunk 4]
```diff
@@ -54,3 +54,26 @@ type ImageSecurityPolicyList struct {
...
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AttestationAuthority struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	NoteReference        string   `json:"noteReference"`
+	PrivateKeySecretName []string `json:"privateKeySecretName"`
```
Comment (dlorenc) [inline pkg/kritis/apis/kritis/v1beta1/types.go]: Why are these repeated?

Comment (aaron-prindle) [reply pkg/kritis/apis/kritis/v1beta1/types.go]: i have changed all elements to be string now, is that what you are referring to?

**Pull request merged**